### PR TITLE
feat(helm): add --password-stdin to `helm repo add`

### LINKED ISSUE #10032: `helm repo add` is missing --password-stdin
**What**

You currently can't pipe the password into `helm repo add`:

```
$ echo password | helm repo add repo-name https://repo-url --username username
Password:
Error: inappropriate ioctl for device
```

I would expect the option `--password-stdin` to be available:

```
$ echo password | helm repo add repo-name https://repo-url --username username --password-stdin
"repo-name" has been added to your repositories
```

**Why**

There are two reasons I see for adding this:

* I personally would expect that it's possible to pipe the password into
  `helm repo add` but that's currently not the case. If I understand it
  correctly, you currently either need to pass the password via a cli
  parameter (`--password`) or use a expect/send mechanism. The
  cli parameter mechanism also isn't secure as it might expose the
  password via the process name.

* Subcommands like `helm registry login` already support
  `--password-stdin`. The cli interfaces should be consistent regarding
  which options they support.

### PR DESCRIPTION
**Pull request by schra**
**What**

Without this commit, you can't pipe the password into `helm repo add`:

```
$ echo password | helm repo add repo-name https://repo-url --username username
Password:
Error: inappropriate ioctl for device
```

This commit introduces `--password-stdin`:

```
$ echo password | helm repo add repo-name https://repo-url --username username --password-stdin
"repo-name" has been added to your repositories
```

Fixes #10032

**Why**

There are two reasons I see for adding this:

* I personally would expect that it's possible to pipe the password into
  `helm repo add` but that's currently not the case. If I understand it
  correctly, you currently either need to pass the password via a cli
  parameter (`--password`) or use a expect/send mechanism.

* Subcommands like `helm registry login` already support
  `--password-stdin`. The cli interfaces should be consistent regarding
  which options they support.

**Notes**

* I basically just copy-pasted code from `cmd/helm/registry_login.go`.

* I don't know if this is the best approach to enable piping into the command. Maybe it would be sufficient to replace `term.ReadPassword(fd)` with something else that also supports piping/ doesn't yield `inappropriate ioctl for device`.

**Special notes for your reviewer**:

**If applicable**:
- [ ] this PR contains documentation
- [ ] this PR contains unit tests
- [ ] this PR has been tested for backwards compatibility

### CODE REVIEW

**Comment by cndoit18** [comment]
PRs must be linked to an existing issue. If none exists, may you please open one explaining the issue and expected outcome?
Please also write tests to test and verify this PR fixes your issue. Thanks.

File: cmd/helm/repo_add.go [hunk 1]
```diff
@@ -143,14 +145,24 @@ func (o *repoAddOptions) run(out io.Writer) error {
...
 
 	if o.username != "" && o.password == "" {
-		fd := int(os.Stdin.Fd())
-		fmt.Fprint(out, "Password: ")
-		password, err := term.ReadPassword(fd)
-		fmt.Fprintln(out)
-		if err != nil {
-			return err
+		if o.passwordFromStdinOpt {
+			passwordFromStdin, err := ioutil.ReadAll(os.Stdin)
```
Comment (cndoit18) [inline cmd/helm/repo_add.go:149]: Is it possible to use `fmt.Scanln` here instead, or use the `io.ReadAll` instead?

**Review by cndoit18 — COMMENTED** [review]
Could you add some unit tests?

File: cmd/helm/repo_add.go (hunk 1 above)
Comment (schra) [reply cmd/helm/repo_add.go]: Done

**Review by technosophos — APPROVED** [review]
Code LGTM

**Review by yxxhero — APPROVED** [review]
LGTM

**Review by bacongobbler — APPROVED** [review]

**Pull request merged**